tools/wake-format/wake-format.wake

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 SiFive, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You should have received a copy of LICENSE.Apache2 along with
+# this software. If not, you may obtain a copy at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package build_wake
+from wake import _
+
+target buildWakeFormat variant =
+    tool here Nil variant "bin/wake-format" (parser, gopt, util,)
```

Comment:
One thing to be aware of is that `tool` isn't creating `bin/wake-format` until `make install` is run, it's creating `wake-format.native-cpp14-release`.
I know that there are other opinions about this topic, but I find it nice when we're able to use the build output from the source tree without always having to do the install step.

Reply:
yeah, makes sense. I've just been running `./bin/wake-format.native-cpp14-release ...` but I can see how that wouldn't be ideal.

I personally don't have a strong opinion but it sounds like some other folks do. Would be happy to take a look at that but seems better for a separate discussion/PR

Reply:
I'm honestly not a big fan of the whole "native-cpp14-release" thing but I don't have strong opinions either. The only benefit is that you can do debug and non-debug builds at the same time. Currently `share` has to be relative to the wake binary in order for wake to find the standard library otherwise I'd suggest putting wake in `bin/native-cpp14-release/wake` and `bin/native-cpp14-debug/wake` but then we'd need to change how wake finds the standard library. We could solve that having `native-cpp14-debug/bin/wake` and `native-cpp14-release/bin/wake` and then symlink the share dir into `native-cpp14-debug` and `native-cpp14-release` I suppose. Then an "install" would just be a copy that resolved all symlinks.

I'm not 100% sure how wake should find the standard library but I'm not a fan of how its done now anyway.